File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:

        my_dict = {}

        result = []

        for num in nums:

            if num not in my_dict:

                my_dict[num] = 1
            
            else:

                my_dict[num] += 1

        
        while(k != 0):

            result.append(max(my_dict, key = my_dict.get))

            del my_dict[max(my_dict, key = my_dict.get)]

            k -= 1;
        

        return result;

            
    
        

File: Data Structures & Algorithms/daily-temperatures/submission-1.cpp
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& temperatures) {

        vector<int> result(temperatures.size());

        for (int i = 0; i < temperatures.size(); i++) {

            int j = i + 1;

            while(j < temperatures.size() && temperatures[j] <= temperatures[i]) {

                j++;
            }

            if(j < temperatures.size()) {

                result[i] = j - i;
            } else {

                result[i] = 0;
            }

        }

        return result; 
        
    }
};
